node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -75,37 +78,41 @@ case class DataMessageHandler(
         }
       case filter: CompactFilterMessage =>
         logger.debug(s"Received ${filter.commandName}, $filter")
+        val batchSizeFull: Boolean = currentFilterBatch.size == chainConfig.filterBatchSize - 1
         for {
-          (newCount, newSyncing) <- if (receivedFilterCount == chainConfig.filterBatchSize - 1) {
+          (newBatch, newSyncing) <- if (batchSizeFull) {
             logger.info(
               s"Received maximum amount of filters in one batch. This means we are not synced, requesting more")
             for {
               _ <- sendNextGetCompactFilterCommand(peerMsgSender,
                                                    filter.blockHash.flip)
-            } yield (0, syncing)
+            } yield (Vector.empty, syncing)
           } else {
             for {
-              filterHeaderCount <- chainApi.getFilterHeaderCount
-              filterCount <- chainApi.getFilterCount
+              filterHeaderCount <- chainApi.getFilterHeaderCount()
+              filterCount <- chainApi.getFilterCount()
             } yield {
               val syncing = filterCount < filterHeaderCount - 1
               if (!syncing) {
                 logger.info(s"We are synced")
               }
-              (receivedFilterCount + 1, syncing)
+              (currentFilterBatch :+ filter, syncing)
             }
           }
           newChainApi <- chainApi.processFilter(filter)
-          blockFilter <- Future(
-            BlockFilter.fromBytes(filter.filterBytes, filter.blockHash))
-          _ <- callbacks.executeOnCompactFilterReceivedCallbacks(
-            logger,
-            filter.blockHash,
-            blockFilter)
+          // If we are not syncing or our filter batch is full, process the filters
+          _ <- if (!syncing || batchSizeFull) {
```

Comment:
Initially I implemented compact filters sync this. And I reverted it because 1) it didn't really speed up the sync process; 2) the node was able to sync only a multiple of `batchSizeFull` filter. So if the batch size is 100 and we initially synced 101 filters, we'll process the the 102nd filter only in ~approximately  16 hours, when the next batch is filled up.

Reply:
That's why we do the check on `syncing`. Then it will check every filter immediately after it initially catches up to the chain tip